---
title: "Exploring cleaned ADNI data"
output: github_document
---


```{r setup}
library(tidyverse)
library(ggplot2)
library(readxl)
library(RColorBrewer)

library(pheatmap)
library(polycor)
library(corrplot)
library(factoextra)

library(sva)

library(randomForest)
library(caret)
library(tree)
library(Metrics)
library(e1071)
library(Boruta)

# devtools::install_github("plotly/plotly.R")
library(plotly)
library(cowplot)

library(limma)
```


```{r}
# check datasets to be used
# adni_gene_expression_subjectInfo_allInfo %>% head # subject information
# adni_gene_expression_values %>% head # gene expression values
```


```{r}
# remove subject(s) with diagnosis as NA
# which(is.na(adni_gene_expression_subjectInfo_allInfo$ADNIMERGE_Dx)) #subject 679
# 
# gene_expression_values_toAnalyse <- adni_gene_expression_values[-c(679),]
# subjectInfo_toAnalyse <- adni_gene_expression_subjectInfo_allInfo[-c(679),]
```


```{r}
subjectInfo_toAnalyse_selectRIN
gene_expression_values_toAnalyse_selectRIN_selectProbes
```


## PCA for possible batch effects

```{r}
object <- gene_expression_values_toAnalyse 

# code for pca and scree plot adapted from getMethod("plotPCA", "DESeqTransform")
res.pca <- prcomp(gene_expression_values_toAnalyse)
percentVar <- res.pca$sdev^2/sum(res.pca$sdev^2)
```


```{r fig.width = 7, fig.height = 5}
# scree plot
fviz_eig(res.pca)
```


### RIN:

```{r fig.width = 10, fig.height = 6}
## coloured by RIN
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][2]

# subjectInfo_toAnalyse %>% colnames

RIN <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                RIN = RIN,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_RIN <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = RIN)) + 
  scale_color_continuous(guide = guide_colorbar(
    title = "RIN", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_RIN <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = RIN)) + 
  scale_color_continuous(guide = guide_colorbar(
    title = "RIN", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_RIN <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by RIN)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_RIN <- plot_grid(
    ADNI_allGenes_PC1_2_RIN, 
    ADNI_allGenes_PC2_3_RIN, 
    rel_widths = c(1,0.95),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_RIN.title <- plot_grid(
    title_PCA_ADNI_allGenes_RIN, 
    PCA_ADNI_allGenes_RIN, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_RIN.title
```



### Affy plate:

```{r fig.width = 10, fig.height = 5}
## coloured by Affy plate
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][3]

Affy.Plate <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                Affy.Plate = Affy.Plate,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_Affy.Plate <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = Affy.Plate)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Affy.Plate", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_Affy.Plate <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = Affy.Plate)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Affy.Plate", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_Affy.Plate <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by Affy Plate)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_Affy.Plate <- plot_grid(
    ADNI_allGenes_PC1_2_Affy.Plate, 
    ADNI_allGenes_PC2_3_Affy.Plate, 
    rel_widths = c(1,1.05),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_Affy.Plate.title <- plot_grid(
    title_PCA_ADNI_allGenes_Affy.Plate, 
    PCA_ADNI_allGenes_Affy.Plate, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_Affy.Plate.title
```



### Year of collection:


```{r fig.width = 10, fig.height = 5}
## coloured by year of collection
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][4]

YearofCollection <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                YearofCollection = YearofCollection,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_YearofCollection <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = YearofCollection)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "YearofCollection", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_YearofCollection <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = YearofCollection)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "YearofCollection", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_YearofCollection <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by year of collection)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_YearofCollection <- plot_grid(
    ADNI_allGenes_PC1_2_YearofCollection, 
    ADNI_allGenes_PC2_3_YearofCollection, 
    rel_widths = c(1,1.25),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_YearofCollection.title <- plot_grid(
    title_PCA_ADNI_allGenes_YearofCollection, 
    PCA_ADNI_allGenes_YearofCollection, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_YearofCollection.title
```


### ADNI phase:

```{r fig.width = 10, fig.height = 4}
## coloured by ADNI phase
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][1]

Phase <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                Phase = Phase,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_Phase <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = Phase)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Phase", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_Phase <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = Phase)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Phase", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_Phase <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by ADNI phase)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_Phase <- plot_grid(
    ADNI_allGenes_PC1_2_Phase, 
    ADNI_allGenes_PC2_3_Phase, 
    rel_widths = c(1,1),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_Phase.title <- plot_grid(
    title_PCA_ADNI_allGenes_Phase, 
    PCA_ADNI_allGenes_Phase, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_Phase.title
```


### Diagnosis

```{r fig.width = 10, fig.height = 4}
## coloured by diagnosis at time of gene expression data collection
subjectInfo_toAnalyse$ADNIMERGE_Dx <- factor(subjectInfo_toAnalyse$ADNIMERGE_Dx)
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][5]

Diagnosis <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                Diagnosis = Diagnosis,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_Diagnosis <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = ADNIMERGE_Dx)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Diagnosis", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_Diagnosis <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = ADNIMERGE_Dx)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Diagnosis", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_Diagnosis <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by diagnosis)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_Diagnosis <- plot_grid(
    ADNI_allGenes_PC1_2_Diagnosis, 
    ADNI_allGenes_PC2_3_Diagnosis, 
    rel_widths = c(1,1.3),
    labels = "AUTO", label_size = 15, scale = .9)

PCA_ADNI_allGenes_Diagnosis.title <- plot_grid(
    title_PCA_ADNI_allGenes_Diagnosis, 
    PCA_ADNI_allGenes_Diagnosis, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_Diagnosis.title
```




### Age:

```{r fig.width = 10, fig.height = 6}
## coloured by Age
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][6]

# subjectInfo_toAnalyse %>% colnames

Age <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                Age = Age,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_Age <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = Age)) + 
  scale_color_continuous(guide = guide_colorbar(
    title = "RIN", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_Age <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = Age)) + 
  scale_color_continuous(guide = guide_colorbar(
    title = "RIN", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_Age <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by age)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_Age <- plot_grid(
    ADNI_allGenes_PC1_2_Age, 
    ADNI_allGenes_PC2_3_Age, 
    rel_widths = c(1,0.95),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_Age.title <- plot_grid(
    title_PCA_ADNI_allGenes_Age, 
    PCA_ADNI_allGenes_Age, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_Age.title
```







### Diagnosis at baseline:

```{r fig.width = 10, fig.height = 4}
## coloured by diagnosis at baseline
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][7]

DX_bl <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                DX_bl = DX_bl,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_DX_bl <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = DX_bl)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "DX_bl", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_DX_bl <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = DX_bl)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "DX_bl", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_DX_bl <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by diagnosis at baseline)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_DX_bl <- plot_grid(
    ADNI_allGenes_PC1_2_DX_bl, 
    ADNI_allGenes_PC2_3_DX_bl, 
    rel_widths = c(1,1),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_DX_bl.title <- plot_grid(
    title_PCA_ADNI_allGenes_DX_bl, 
    PCA_ADNI_allGenes_DX_bl, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_DX_bl.title
```



### PTGENDER:

```{r fig.width = 10, fig.height = 4}
## coloured by gender
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][8]

PTGENDER <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                PTGENDER = PTGENDER,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_PTGENDER <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = PTGENDER)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "PTGENDER", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_PTGENDER <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = PTGENDER)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "PTGENDER", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_PTGENDER <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by patient gender)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_PTGENDER <- plot_grid(
    ADNI_allGenes_PC1_2_PTGENDER, 
    ADNI_allGenes_PC2_3_PTGENDER, 
    rel_widths = c(1,1),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_PTGENDER.title <- plot_grid(
    title_PCA_ADNI_allGenes_PTGENDER, 
    PCA_ADNI_allGenes_PTGENDER, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_PTGENDER.title
```



### Number of APOE4 alleles:

```{r fig.width = 10, fig.height = 4}
## coloured by number of APOE4 alleles
subjectInfo_toAnalyse$APOE4 <- factor(subjectInfo_toAnalyse$APOE4)
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][9]

APOE4 <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                APOE4 = APOE4,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_APOE4 <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = APOE4)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "APOE4", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_APOE4 <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = APOE4)) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "APOE4", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_APOE4 <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by number of APOE4 alleles)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_APOE4 <- plot_grid(
    ADNI_allGenes_PC1_2_APOE4, 
    ADNI_allGenes_PC2_3_APOE4, 
    rel_widths = c(1,1),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_APOE4.title <- plot_grid(
    title_PCA_ADNI_allGenes_APOE4, 
    PCA_ADNI_allGenes_APOE4, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_APOE4.title
```





### Race:

```{r fig.width = 10, fig.height = 4}
## coloured by race
# subjectInfo_toAnalyse$APOE4 <- factor(subjectInfo_toAnalyse$APOE4)
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][10]

Race <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                Race = Race,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_Race <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = Race), alpha = .4) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Race", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_Race <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = Race), alpha = .4) + 
  #scale_color_continuous(guide = guide_colorbar(
  #  title = "Race", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_Race <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by race)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_Race <- plot_grid(
    ADNI_allGenes_PC1_2_Race, 
    ADNI_allGenes_PC2_3_Race, 
    rel_widths = c(1,1.35),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_Race.title <- plot_grid(
    title_PCA_ADNI_allGenes_Race, 
    PCA_ADNI_allGenes_Race, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_Race.title
```



### MMSE:

```{r fig.width = 10, fig.height = 6}
## coloured by MMSE
intgroup.df <- subjectInfo_toAnalyse[, c("Phase", "RIN", "Affy.Plate", "YearofCollection", 
                                         "ADNIMERGE_Dx", "Age", "DX_bl", "PTGENDER", "APOE4", "Race", 
                                         "MMSE")][11]

# subjectInfo_toAnalyse %>% colnames

MMSE <- intgroup.df

d <- data.frame(PC1 = res.pca$x[,1], PC2 = res.pca$x[, 2], PC3 = res.pca$x[, 3], 
                MMSE = MMSE,
                intgroup.df, name = rownames(object))

## Plot of PC1 and PC2
ADNI_allGenes_PC1_2_MMSE <- 
  ggplot(data = d, aes(x = PC1, y = PC2)) +
  geom_point(size = 1, aes(color = MMSE), alpha = .5) + 
  scale_color_continuous(guide = guide_colorbar(
    title = "RIN", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC1: ", round(percentVar[1] * 100), "% variance")) +
  ylab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC1 and PC2") +
  theme(text = element_text(size = 15), 
       legend.position = "none") 

## Plot of PC2 and PC3
ADNI_allGenes_PC2_3_MMSE <- 
  ggplot(data = d, aes(x = PC2, y = PC3)) +
  geom_point(size = 1, aes(color = MMSE), alpha = .5) + 
  scale_color_continuous(guide = guide_colorbar(
    title = "RIN", title.position = "top", title.vjust = 1)) +
  xlab(paste0("PC2: ", round(percentVar[2] * 100), "% variance")) +
  ylab(paste0("PC3: ", round(percentVar[3] * 100), "% variance")) +
  coord_fixed() + 
  ggtitle("PC2 and PC3") + 
  theme(text = element_text(size = 15), 
       legend.position = "right")
```


```{r fig.width=10, fig.height=4}
title_PCA_ADNI_allGenes_MMSE <- ggdraw() + 
  draw_label(
    "PCA for all genes from ADNI (coloured by MMSE)",
    fontface = 'bold',
    x = 0,
    hjust = 0,
      size = 20
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = ggplot2::margin(0, 0, 0, 7)
  )


PCA_ADNI_allGenes_MMSE <- plot_grid(
    ADNI_allGenes_PC1_2_MMSE, 
    ADNI_allGenes_PC2_3_MMSE, 
    rel_widths = c(1,0.95),
    labels = "AUTO", label_size = 15, scale = 0.9)

PCA_ADNI_allGenes_MMSE.title <- plot_grid(
    title_PCA_ADNI_allGenes_MMSE, 
    PCA_ADNI_allGenes_MMSE, 
    ncol = 1,
    rel_heights = c(0.1, 1))

PCA_ADNI_allGenes_MMSE.title
```


```{r}
sessionInfo()
```



