---
title: "20230130 cleaning up ADNI raw data"
output: github_document
---

```{r setup}
library(tidyverse)
library(ggplot2)
library(readxl)
library(RColorBrewer)

library(pheatmap)
library(polycor)
library(corrplot)
library(factoextra)

library(sva)

library(randomForest)
library(caret)
library(tree)
library(Metrics)
library(e1071)
library(Boruta)

# devtools::install_github("plotly/plotly.R")
library(plotly)
library(cowplot)

library(limma)

# install.packages("here")
library(here)

knitr::opts_knit$set(root.dir = "~/Desktop/NTU/EG_Lab/Jan2023_ADNI_analysis")
```



Reading in files:

```{r}
## main gene expression file
adni_gene_expression_raw <- read.csv("../data/ADNI_raw_files/ADNI_Gene_Expression_Profile/ADNI_Gene_Expression_Profile.csv", header = FALSE)
# adni_gene_expression_raw %>% head
adni_gene_expression_raw[9:15,1:10]
```


```{r}
### first 8 rows of the dataframe are information about the subjects
adni_gene_expression_subjectInfo <- adni_gene_expression_raw[1:8,c(1,4:ncol(adni_gene_expression_raw))] %>%
  column_to_rownames(var = "V1")
colnames(adni_gene_expression_subjectInfo) <- c(paste0("subject_", 1:ncol(adni_gene_expression_subjectInfo)))
adni_gene_expression_subjectInfo[,(ncol(adni_gene_expression_subjectInfo)-5):ncol(adni_gene_expression_subjectInfo)]
# last column is empty (used to store gene symbol details in original df)
adni_gene_expression_subjectInfo <- adni_gene_expression_subjectInfo[,1:(ncol(adni_gene_expression_subjectInfo)-1)]
adni_gene_expression_subjectInfo <- adni_gene_expression_subjectInfo %>% t %>% data.frame #transpose for easier cleaning

adni_gene_expression_subjectInfo %>% nrow # 744

adni_gene_expression_subjectInfo$RIN <- as.numeric(adni_gene_expression_subjectInfo$RIN)
adni_gene_expression_subjectInfo$Phase <- factor(adni_gene_expression_subjectInfo$Phase)
adni_gene_expression_subjectInfo$Affy.Plate <- factor(adni_gene_expression_subjectInfo$Affy.Plate)
adni_gene_expression_subjectInfo$YearofCollection <- factor(adni_gene_expression_subjectInfo$YearofCollection)

# adni_gene_expression_subjectInfo %>% head
```


```{r}
### collate the rest of the rows of the dataframe to make gene expression matrix
adni_gene_expression_values <- adni_gene_expression_raw[9:nrow(adni_gene_expression_raw),]

# rename columns to match subjects and additional information
colnames(adni_gene_expression_values) <- c("ProbeSet", "LocusLink", "Symbol", 
                                           c(paste0("subject_", 1:(ncol(adni_gene_expression_values)-4))),
                                           "geneInfo")
# adni_gene_expression_values[(ncol(adni_gene_expression_values)-5):ncol(adni_gene_expression_values)]

# remove first row (column headings already added)
adni_gene_expression_values <- adni_gene_expression_values[2:nrow(adni_gene_expression_values), ]

# change microarray values to all numeric
adni_gene_expression_values[,4:(ncol(adni_gene_expression_values)-1)] <- lapply(adni_gene_expression_values[,4:(ncol(adni_gene_expression_values)-1)], as.numeric)
# adni_gene_expression_values %>% head(20)

# fix gene name for SEPT2
which(adni_gene_expression_values$Symbol == "2-Sep")
adni_gene_expression_values$Symbol[267] <- "SEPT2"
adni_gene_expression_values$Symbol[268] <- "SEPT2"
adni_gene_expression_values$Symbol[34688] <- "SEPT2"
adni_gene_expression_values$Symbol[49209] <- "SEPT2"
```


```{r}
# separate gene expression dataframe into probe/gene information and gene expression values
adni_gene_expression_geneInfo <- adni_gene_expression_values[,c(1:3,ncol(adni_gene_expression_values))]

# keep ProbeSet as rownames (most complete set of gene "names", remove all other geneInfo columns)
adni_gene_expression_values <- adni_gene_expression_values[,c(1, 4:(ncol(adni_gene_expression_values)-1))] 
rownames(adni_gene_expression_values) <- adni_gene_expression_values$ProbeSet
adni_gene_expression_values <- adni_gene_expression_values[,c(2:ncol(adni_gene_expression_values))] %>% 
  t %>%
  as.data.frame()
# adni_gene_expression_values %>% head
adni_gene_expression_values %>% nrow # 744

# save(adni_gene_expression_values, file = "adni_gene_expression_values.RData")
```


```{r}
# import diagnosis data
diagnosis_info <- read.csv("~/Desktop/NTU/EG_Lab/data/ADNI_raw_files/Diagnosis/DXSUM_PDXCONV_ADNIALL.csv")
# diagnosis_info %>% head

diagnosis_info.split_phase <- diagnosis_info %>% split(diagnosis_info$Phase)
# diagnosis_info.split_phase$ADNIGO %>% head()
# diagnosis_info.split_phase$ADNI2
```

ADNI2:
DXCHANGE	
DXSUM	
Diagnostic Summary	1.  Which best describes the participant's change in cognitive status from last visit to current visit?  At Screening Visit, indicate initial diagnosis using one of the 'Stable' options.	N	1	
1=Stable: NL; 
2=Stable: MCI; 
3=Stable: Dementia; 
4=Conversion: NL to MCI; 
5=Conversion: MCI to Dementia; 
6=Conversion: NL to Dementia; 
7=Reversion: MCI to NL; 
8=Reversion: Dementia to MCI; 
9=Reversion: Dementia to NL

ADNIGO:
ADNIGO	
DXCHANGE	
DXSUM	Diagnostic Summary	1.  Which best describes the participant's change in cognitive status from last visit to current visit:	N	1	
1=Stable: NL to NL; 
2=Stable: MCI to MCI; 
3=Stable: Dementia to Dementia; 
4=Conversion: NL to MCI; 
5=Conversion: MCI to Dementia; 
6=Conversion: NL to Dementia; 
7=Reversion: MCI to NL; 
8=Reversion: Dementia to MCI; 
9=Reversion: Dementia to NL



```{r}
# adni_gene_expression_subjectInfo %>% head
adni_gene_expression_subjectInfo.split_phase <- adni_gene_expression_subjectInfo %>% 
  split(adni_gene_expression_subjectInfo$Phase)
# adni_gene_expression_subjectInfo.split_phase
```


### Diagnosis information for adni2

```{r}
# list of adni2 diagnosis split by subject
#adni2_diagnosis.split_subject <- split(diagnosis_info.split_phase$ADNI2, diagnosis_info.split_phase$ADNI2$PTID) 
# adni2_diagnosis.split_subject$"002_S_0295"
# which(adni2_diagnosis.split_subject$"002_S_0295"$VISCODE == "v06")

# initialise empty vector to store diagnosis values
gene_expression_adni2_diagnosis_values <- rep(0,nrow(adni_gene_expression_subjectInfo.split_phase$ADNI2))
names(gene_expression_adni2_diagnosis_values) <- rownames(adni_gene_expression_subjectInfo.split_phase$ADNI2)

for (i in 1:nrow(adni_gene_expression_subjectInfo.split_phase$ADNI2)) {
  
  subjectID_i <- adni_gene_expression_subjectInfo.split_phase$ADNI2$SubjectID[i]
  visit_i <- adni_gene_expression_subjectInfo.split_phase$ADNI2$Visit[i]
  
  # find index of subject 
  adni2_diagnosis.subject_index_i <- which(diagnosis_info.split_phase$ADNI2$PTID == subjectID_i)
  # find index of visit
  adni2_diagnosis.visit_index_i <- which(diagnosis_info.split_phase$ADNI2$VISCODE == visit_i)
  
  # select row based on visit
  adni2_diagnosis.subject_visit_i <-
    diagnosis_info.split_phase$ADNI2[intersect(adni2_diagnosis.subject_index_i, adni2_diagnosis.visit_index_i),]
  
  # get DXCHANGE column value
  diagnosis_value_i <- adni2_diagnosis.subject_visit_i$DXCHANGE
  
  if(length(diagnosis_value_i) == 0){
    diagnosis_value_i = NA
  }
  
  gene_expression_adni2_diagnosis_values[i] <- diagnosis_value_i # store diagnosis value into vector
}

```

```{r}
# vector of subjects that have no diagnosis values
which(is.na(gene_expression_adni2_diagnosis_values))

 # subject_73  subject_79 subject_124 subject_210 subject_361 subject_386 subject_540 subject_636 subject_679 
 #         47          52          76         135         223         243         335         385         407

# check DXSUM_PDCONV_ADNIALL sheet for individual patient information

adni_gene_expression_subjectInfo.split_phase$ADNI2[47,] # 072_S_4102 v04
# conversion to dementia at visit 11 but still MCI at visit 3 and visit 5
# assumption of MCI at visit 4
gene_expression_adni2_diagnosis_values[47] <- 2

adni_gene_expression_subjectInfo.split_phase$ADNI2[52,] # 006_S_4192 v04
# status is 3 at visit 1, 3, 5 and 11
# assumption of status 3 at visit 4
gene_expression_adni2_diagnosis_values[52] <- 3

adni_gene_expression_subjectInfo.split_phase$ADNI2[76,] # 013_S_4268 v04
# status is 2 at visit 1, 3, 5, conversion to 7 at visit 11 (reversion to NL)
# assumption of status 2 at visit 4
gene_expression_adni2_diagnosis_values[76] <- 2

adni_gene_expression_subjectInfo.split_phase$ADNI2[135,] # 072_S_4103 v04
# status is 1 at visit 1, 3, 5
# assumption of status 1 at visit 4
gene_expression_adni2_diagnosis_values[135] <- 1

adni_gene_expression_subjectInfo.split_phase$ADNI2[223,] # 009_S_4564 v02
# status is 3 at visit 1 and 3
# assumption of status 2 at visit 2
gene_expression_adni2_diagnosis_values[223] <- 2

adni_gene_expression_subjectInfo.split_phase$ADNI2[243,] # 022_S_4266 v04
# status is 1 at visit 1, 3 and 5
# assumption of status 1 at visit 4
gene_expression_adni2_diagnosis_values[243] <- 1

adni_gene_expression_subjectInfo.split_phase$ADNI2[335,] # 114_S_4379 v04
# status is 3 at visit 1, 3 and 5
# assumption of status 3 at visit 4
gene_expression_adni2_diagnosis_values[335] <- 3

adni_gene_expression_subjectInfo.split_phase$ADNI2[385,] # 094_S_4282 v04
# status is 3 at visit 1, 3 and 5
# assumption of status 3 at visit 4
gene_expression_adni2_diagnosis_values[385] <- 3

adni_gene_expression_subjectInfo.split_phase$ADNI2[407,] # 037_S_4432 v04
# status is 2 at visit 1 and 3, status is 5 at visit 5 (conversion to dementia)
# remove subject from later analysis (tag as NA for removal later)
gene_expression_adni2_diagnosis_values[407] <- NA

gene_expression_adni2_diagnosis_values %>% table
#   1   2   3   5 
# 171 212  61   5 

# gene_expression_adni2_diagnosis_values %>% head
```


```{r}
# new adni2 subject info including diagnosis
adni2_diagnosis_subjectInfo <- adni_gene_expression_subjectInfo.split_phase$ADNI2 %>%
  mutate(Diagnosis = gene_expression_adni2_diagnosis_values,
         .after = "SubjectID")
# adni2_diagnosis_subjectInfo %>% head
```



## Diagnosis information for ADNIGO

```{r}
# initialise empty vector to store diagnosis values
gene_expression_ADNIGO_diagnosis_values <- rep(0,nrow(adni_gene_expression_subjectInfo.split_phase$ADNIGO))
names(gene_expression_ADNIGO_diagnosis_values) <- rownames(adni_gene_expression_subjectInfo.split_phase$ADNIGO)

for (i in 1:nrow(adni_gene_expression_subjectInfo.split_phase$ADNIGO)) {
  
  subjectID_i <- adni_gene_expression_subjectInfo.split_phase$ADNIGO$SubjectID[i]
  visit_i <- adni_gene_expression_subjectInfo.split_phase$ADNIGO$Visit[i]
  
  # find index of subject 
  ADNIGO_diagnosis.subject_index_i <- which(diagnosis_info.split_phase$ADNIGO$PTID == subjectID_i)
  # find index of visit
  ADNIGO_diagnosis.visit_index_i <- which(diagnosis_info.split_phase$ADNIGO$VISCODE == visit_i)
  
  # select row based on visit
  ADNIGO_diagnosis.subject_visit_i <-
    diagnosis_info.split_phase$ADNIGO[intersect(ADNIGO_diagnosis.subject_index_i, ADNIGO_diagnosis.visit_index_i),]
  
  # get DXCHANGE column value
  diagnosis_value_i <- ADNIGO_diagnosis.subject_visit_i$DXCHANGE
  
  if(length(diagnosis_value_i) == 0){
    diagnosis_value_i = NA
  }
  
  gene_expression_ADNIGO_diagnosis_values[i] <- diagnosis_value_i # store diagnosis value into vector
}
```


```{r}
which(is.na(gene_expression_ADNIGO_diagnosis_values)) 
# subject_60 
#         22 

# check DXSUM_PDCONV_ADNIALL sheet for individual patient information

adni_gene_expression_subjectInfo.split_phase$ADNIGO[22,] # 073_S_2182 m03
# status at bl is 2 and m06 is 2
# assumption of status 2 at m03
gene_expression_ADNIGO_diagnosis_values[22] <- 2

gene_expression_ADNIGO_diagnosis_values %>% table
 #  1   2   3   4   5   7   8 
 # 73 159  37  10  13   2   1 

# gene_expression_ADNIGO_diagnosis_values %>% head
```




```{r}
# new adnigo subject info including diagnosis
adnigo_diagnosis_subjectInfo <- adni_gene_expression_subjectInfo.split_phase$ADNIGO %>%
  mutate(Diagnosis = gene_expression_ADNIGO_diagnosis_values,
         .after = "SubjectID")
# adnigo_diagnosis_subjectInfo %>% head
```

### Combine patient info tables from ADNI2 and ADNIGO (with added diagnosis column)

```{r}
adni_gene_expression_subjectInfo_Dx <- rbind(adni2_diagnosis_subjectInfo, adnigo_diagnosis_subjectInfo)

# reorder rows
adni_gene_expression_subjectInfo_Dx <- adni_gene_expression_subjectInfo_Dx[
  match(rownames(adni_gene_expression_values), 
        rownames(adni_gene_expression_subjectInfo_Dx)),]

# adni_gene_expression_subjectInfo_Dx %>% head

adni_gene_expression_subjectInfo_Dx$Diagnosis %>% table
```


```{r}
## diagnosis dictionary file
adni_diagnosis_metadata_raw <- read.csv("../data/ADNI_raw_files/Diagnosis/DXSUM_PDXCONV_ADNIALL.csv")
# adni_diagnosis_metadata_raw %>% head(10)
```

```{r}
write.csv(adni_gene_expression_values, file = "20230201_adni_gene_expression_values.csv")
write.csv(adni_gene_expression_subjectInfo_Dx, file = "20230201_adni_gene_expression_subjectInfo_Dx.csv")
```



## Compiling further patient information (age, sex, MMSE scores)

```{r}
ADNIMERGE <- read.csv("/Users/ohsherli/Desktop/NTU/EG_Lab/data/ADNI_raw_files/Data___Database/ADNIMERGE.csv")
APOERES <- read.csv("/Users/ohsherli/Desktop/NTU/EG_Lab/data/ADNI_raw_files/APOERES.csv")
ADNI_mRNA_Age <- read.csv("/Users/ohsherli/Desktop/NTU/EG_Lab/data/ADNI_raw_files/ADNI_mRNA_Age.csv")
```



**From APOERES_DICT:**
APGEN1	APOEGO2	ADNI GO & 2 APOE Genotypes	-4	Genotype - Allele 1
APGEN2	APOEGO2	ADNI GO & 2 APOE Genotypes	-4	Genotype - Allele 2


```{r}
# save.image(file = "20230201_workspace_cleaning_adni.RData")
```


```{r}
sessionInfo()
```



