---
title: "Using ADNIMERGE diagnosis labels and comparing between labels"
output: html_notebook
---

```{r setup}
library(tidyverse)
library(ggplot2)
library(readxl)
library(RColorBrewer)

library(pheatmap)
library(polycor)
library(corrplot)

library(sva)

library(randomForest)
library(caret)
library(tree)
library(Metrics)
library(e1071)
library(Boruta)

# devtools::install_github("plotly/plotly.R")
library(plotly)

# BiocManager::install("sva")
library(sva)

library(stringr)
```


Reading in files:

```{r}
## main gene expression file
adni_gene_expression_raw <- read.csv("../data/ADNI_raw_files/ADNI_Gene_Expression_Profile/ADNI_Gene_Expression_Profile.csv", header = FALSE)
adni_gene_expression_raw %>% head
adni_gene_expression_raw[9:15,1:10]
```


```{r}
### first 8 rows of the dataframe are information about the subjects
adni_gene_expression_subjectInfo <- adni_gene_expression_raw[1:8,c(1,4:ncol(adni_gene_expression_raw))] %>%
  column_to_rownames(var = "V1")
colnames(adni_gene_expression_subjectInfo) <- c(paste0("subject_", 1:ncol(adni_gene_expression_subjectInfo)))
adni_gene_expression_subjectInfo[,(ncol(adni_gene_expression_subjectInfo)-5):ncol(adni_gene_expression_subjectInfo)]
# last column is empty (used to store gene symbol details in original df)
adni_gene_expression_subjectInfo <- adni_gene_expression_subjectInfo[,1:(ncol(adni_gene_expression_subjectInfo)-1)]
adni_gene_expression_subjectInfo <- adni_gene_expression_subjectInfo %>% t %>% data.frame #transpose for easier cleaning

adni_gene_expression_subjectInfo %>% nrow # 744

adni_gene_expression_subjectInfo$RIN <- as.numeric(adni_gene_expression_subjectInfo$RIN)
adni_gene_expression_subjectInfo$Phase <- factor(adni_gene_expression_subjectInfo$Phase)
adni_gene_expression_subjectInfo$Affy.Plate <- factor(adni_gene_expression_subjectInfo$Affy.Plate)
adni_gene_expression_subjectInfo$YearofCollection <- factor(adni_gene_expression_subjectInfo$YearofCollection)

adni_gene_expression_subjectInfo %>% head
```


```{r}
### collate the rest of the rows of the dataframe to make gene expression matrix
adni_gene_expression_values <- adni_gene_expression_raw[9:nrow(adni_gene_expression_raw),]

# rename columns to match subjects and additional information
colnames(adni_gene_expression_values) <- c("ProbeSet", "LocusLink", "Symbol", 
                                           c(paste0("subject_", 1:(ncol(adni_gene_expression_values)-4))),
                                           "geneInfo")
adni_gene_expression_values[(ncol(adni_gene_expression_values)-5):ncol(adni_gene_expression_values)]

# remove first row (column headings already added)
adni_gene_expression_values <- adni_gene_expression_values[2:nrow(adni_gene_expression_values), ]

# change microarray values to all numeric
adni_gene_expression_values[,4:(ncol(adni_gene_expression_values)-1)] <- lapply(adni_gene_expression_values[,4:(ncol(adni_gene_expression_values)-1)], as.numeric)
adni_gene_expression_values %>% head(20)

# fix gene name for SEPT2
which(adni_gene_expression_values$Symbol == "2-Sep")
adni_gene_expression_values$Symbol[267] <- "SEPT2"
adni_gene_expression_values$Symbol[268] <- "SEPT2"
adni_gene_expression_values$Symbol[34688] <- "SEPT2"
adni_gene_expression_values$Symbol[49209] <- "SEPT2"
```


```{r}
# separate gene expression dataframe into probe/gene information and gene expression values
adni_gene_expression_geneInfo <- adni_gene_expression_values[,c(1:3,ncol(adni_gene_expression_values))]

# keep ProbeSet as rownames (most complete set of gene "names", remove all other geneInfo columns)
adni_gene_expression_values <- adni_gene_expression_values[,c(1, 4:(ncol(adni_gene_expression_values)-1))] 
rownames(adni_gene_expression_values) <- adni_gene_expression_values$ProbeSet
adni_gene_expression_values <- adni_gene_expression_values[,c(2:ncol(adni_gene_expression_values))] %>% 
  t %>%
  as.data.frame()
adni_gene_expression_values %>% head
adni_gene_expression_values %>% nrow # 744
```


```{r}
# import diagnosis info from adnimerge dataset
adnimerge_filePath <- "../data/ADNI_raw_files/Data___Database/ADNIMERGE.csv"
adnimerge <- read.csv(adnimerge_filePath)
adnimerge %>% head
```

# Match ADNImerge subjects with list of subjects from gene expression data by study date

```{r}
# split gene expression data by adni2 and adnigo
adni_gene_expression_subjectInfo %>% head
adni_gene_expression_subjectInfo.split_phase <- adni_gene_expression_subjectInfo %>% 
  split(adni_gene_expression_subjectInfo$Phase)
adni_gene_expression_subjectInfo.split_phase
```

## Get diagnosis for ADNI2 subjects

```{r}
# for adni2, use viscode2 instead of viscode for Visit by adding additional column

## get viscode and viscode2 data from PTDEMOG file
PTDX <- read.csv("~/Desktop/NTU/EG_Lab/data/ADNI_raw_files/Diagnosis/DXSUM_PDXCONV_ADNIALL.csv")
PTDX %>% head()
```


```{r}
# extract patient RID 
?str_pad
PT_RID_info <- PTDX[,c("Phase", "RID", "SITEID", "VISCODE", "VISCODE2")]
PT_RID_info %>% head

# add a column where RID is character instead of numeric
PT_RID_info <- PT_RID_info %>% 
  mutate(
    RID_chr = RID %>% as.character, .before = SITEID
  )

# pad elements of new RID_chr column with zeroes to make total of 4 digits
PT_RID_info$RID_chr <- str_pad(PT_RID_info$RID_chr, 4, pad = "0")
PT_RID_info %>% head

PT_RID_info.splitPhase <- split(PT_RID_info, PT_RID_info$Phase)
PT_RID_info.splitPhase.ADNI2 <- PT_RID_info.splitPhase$ADNI2
```


```{r}
# from ADNI2 gene expression data, add new VISCODE2 column based on RID_chr matched with subjectID and VISCODE
adni_gene_expression_subjectInfo.split_phase.ADNI2 <- adni_gene_expression_subjectInfo.split_phase$ADNI2 

RID_ADNI2_vector <- rep(0, nrow(adni_gene_expression_subjectInfo.split_phase.ADNI2))
RID_ADNI_list <- strsplit(adni_gene_expression_subjectInfo.split_phase.ADNI2$SubjectID, 
                          split = "_")

for (i in 1:length(RID_ADNI_list)){
  RID_ADNI_i <- RID_ADNI_list[[i]][3]
  RID_ADNI2_vector[i] <- RID_ADNI_i
}

adni_gene_expression_subjectInfo.split_phase.ADNI2 <- adni_gene_expression_subjectInfo.split_phase.ADNI2 %>%
  mutate(RID_chr = RID_ADNI2_vector, .after = SubjectID)

adni_gene_expression_subjectInfo.split_phase.ADNI2 %>% head
```


```{r}
# match new ADNI2 data with VISCODE2 in PT_RID_info

# initalize empty vector to store VISCODE2 values
ADNI2_geneExpression_VISCODE2_values <- rep(0, nrow(adni_gene_expression_subjectInfo.split_phase.ADNI2))
names(ADNI2_geneExpression_VISCODE2_values) <- rownames(adni_gene_expression_subjectInfo.split_phase.ADNI2)

for(i in 1:nrow(adni_gene_expression_subjectInfo.split_phase.ADNI2)){
  
  subject_RID_i <- adni_gene_expression_subjectInfo.split_phase.ADNI2$RID[i]
  viscode_i <- adni_gene_expression_subjectInfo.split_phase.ADNI2$Visit[i]
  
  # find index of RID in PT_RID_info
  ADNI2_geneExpression_PTDEMOG_RID_match_i <- which(
    PT_RID_info.splitPhase.ADNI2$RID_chr == subject_RID_i
  )
  # find index of viscode in PT_RID_info
  ADNI2_geneExpression_PTDEMOG_viscode_match_i <- which(
    PT_RID_info.splitPhase.ADNI2$VISCODE == viscode_i
  )
  # find intersection of RID and viscode to give row number
  ADNI2_RIDinfo_RID_viscode_match_i <- intersect(
    ADNI2_geneExpression_PTDEMOG_RID_match_i, 
    ADNI2_geneExpression_PTDEMOG_viscode_match_i
  )
  # select row and viscode2 value based on row number
  viscode2_i <- (PT_RID_info.splitPhase.ADNI2[ADNI2_RIDinfo_RID_viscode_match_i,])$VISCODE2
  
  # for rows with missing values, i.e. no VISCODE2 found, use NA
  if (length(viscode2_i)==0){
    viscode2_i = NA
  }
  
  # replace vector position i with viscode2 value
  ADNI2_geneExpression_VISCODE2_values[i] <- viscode2_i
  
}
```


```{r}
# check results of running loop
ADNI2_geneExpression_VISCODE2_values %>% head
ADNI2_geneExpression_VISCODE2_values %>% unique

# get index of subjects with no viscode2 values
which(is.na(ADNI2_geneExpression_VISCODE2_values))

 # subject_73  subject_79 subject_124 subject_210 subject_361 subject_386 subject_540 subject_636 subject_679 
 #         47          52          76         135         223         243         335         385         407 
```


```{r}
ADNI2_geneExpression_VISCODE2_values %>% head(18)
adni_gene_expression_subjectInfo.split_phase.ADNI2 %>% head(18)
```



```{r}
# match new ADNI2 data with VISCODE2 to ADNIMERGE for diagnosis labels

# split ADNIMERGE data by colprot (phase of study to minimise multiple labels across multiple study phases)
adnimerge_phaseSplit <- adnimerge %>% split(adnimerge$COLPROT)
adnimerge_adni2 <- adnimerge_phaseSplit$ADNI2

adnimerge_adni2 %>% head
```


```{r}
# initialize empty vector to store diagnosis labels for ADNI2 data
ADNI2_geneExpression_Dx_values <- rep(0, nrow(adni_gene_expression_subjectInfo.split_phase.ADNI2))
names(ADNI2_geneExpression_Dx_values) <- rownames(adni_gene_expression_subjectInfo.split_phase.ADNI2)

for(i in 1:nrow(adni_gene_expression_subjectInfo.split_phase.ADNI2)){
  
  # get subject RID to match with ADNIMERGE PTID
  subjectID_i <- adni_gene_expression_subjectInfo.split_phase.ADNI2$SubjectID[i]
  # get viscode2 for subject in gene expression data
  subject_viscode2_i <- ADNI2_geneExpression_VISCODE2_values[i]
  
  # for rows with no missing viscode2 values:
  if(is.na(subject_viscode2_i)){
    subject_ADNIMERGE_Dx_i <- NA
  } else {
    # find index of subjectID in ADNIMERGE data
    adnimerge_subjectID_match_i <- which(adnimerge_adni2$PTID == subjectID_i)
    # find index of viscode2 in ADNIMERGE data
    adnimerge_viscode2_match_i <- which(adnimerge_adni2$VISCODE == subject_viscode2_i)
    # find intersection of index of RID and index of viscode2 to give row number of diagnosis
    adnimerge_subjectID_viscode2_match_i <- intersect(adnimerge_subjectID_match_i, adnimerge_viscode2_match_i)
    # select row and diagnosis according to ADNIMERGE data
    subject_ADNIMERGE_Dx_i <- (adnimerge_adni2[adnimerge_subjectID_viscode2_match_i,])$DX
  }
  
  # store diagnosis value in vector
  ADNI2_geneExpression_Dx_values[i] <- subject_ADNIMERGE_Dx_i
}
```


```{r}
ADNI2_geneExpression_Dx_values %>% head
```


```{r}
# fix the unknown diagnosis values by checking diagnosis before and after the visit

# get index of subjects with no viscode2 values
which(is.na(ADNI2_geneExpression_VISCODE2_values))

 # subject_73  subject_79 subject_124 subject_210 subject_361 subject_386 subject_540 subject_636 subject_679 
 #         47          52          76         135         223         243         335         385         407
```


```{r}
# subject_73
adni_gene_expression_subjectInfo.split_phase.ADNI2[47, ]
# visit v04, subject ID 072_S_4102
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4102")
# [1]  163  351  860 2028
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "072_S_4102"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is MCI, so use MCI as diagnosis
ADNI2_geneExpression_Dx_values[47] <- "MCI"
```


```{r}
# subject_79
adni_gene_expression_subjectInfo.split_phase.ADNI2[52, ]
# visit v04, subject ID 006_S_4192
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4192")
# [1]  355  652 1435 2240 3996
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "006_S_4192"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is Dementia, so use Dementia as diagnosis
ADNI2_geneExpression_Dx_values[52] <- "Dementia"
```


```{r}
# subject_124
adni_gene_expression_subjectInfo.split_phase.ADNI2[76, ]
# visit v04, subject ID 013_S_4268
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4268")
# [1]  471  747 1619 2359 3844 4638
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "013_S_4268"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is MCI, so use MCI as diagnosis
# note reversion back to control from m12
ADNI2_geneExpression_Dx_values[76] <- "MCI"
```


```{r}
# subject_210
adni_gene_expression_subjectInfo.split_phase.ADNI2[135, ]
# visit v04, subject ID 072_S_4103
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4103")
# [1]  160  368 1479 2023 3703
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "072_S_4103"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is CN, so use CN as diagnosis
ADNI2_geneExpression_Dx_values[135] <- "CN"
```


```{r}
# subject_361
adni_gene_expression_subjectInfo.split_phase.ADNI2[223, ]
# visit v02, subject ID 009_S_4564
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4564")
# [1] 1023 4070
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between sc and bl
adnimerge_adni2[which(adnimerge_adni2$PTID == "009_S_4564"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# only diagnosis at bl available, diagnosis is MCI
ADNI2_geneExpression_Dx_values[223] <- "MCI"
```


```{r}
# subject_386
adni_gene_expression_subjectInfo.split_phase.ADNI2[243, ]
# visit v04, subject ID 022_S_4266
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4266")
# [1]  376  704 1589 2719 4048 5530
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "022_S_4266"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is CN, so use CN as diagnosis
ADNI2_geneExpression_Dx_values[243] <- "CN"
```


```{r}
# subject_540
adni_gene_expression_subjectInfo.split_phase.ADNI2[335, ]
# visit v04, subject ID 114_S_4379
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4379")
# [1]  604 1272 2462 3067
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "114_S_4379"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is Dementia, so use Dementia as diagnosis
ADNI2_geneExpression_Dx_values[335] <- "Dementia"
```


```{r}
# subject_636
adni_gene_expression_subjectInfo.split_phase.ADNI2[385, ]
# visit v04, subject ID 094_S_4282
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4282")
# [1]  452  735 1291 3881
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "094_S_4282"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl and m06, diagnosis is Dementia, so use Dementia as diagnosis
ADNI2_geneExpression_Dx_values[385] <- "Dementia"
```


```{r}
# subject_679
adni_gene_expression_subjectInfo.split_phase.ADNI2[407, ]
# visit v04, subject ID 037_S_4432
subject_viscode_check <- which(PT_RID_info.splitPhase.ADNI2$RID_chr == "4432")
# [1]  862 1185 1991 2785 4274 4854 5352
PT_RID_info.splitPhase.ADNI2[subject_viscode_check,]
# viscode2 is between bl and m06
adnimerge_adni2[which(adnimerge_adni2$PTID == "037_S_4432"), c("RID", "PTID", "VISCODE", "DX_bl", "DX")]
# at bl, diagnosis is MCI, but at m06, diagnosis is Dementia
# remove subject from analysis because diagnosis is unclear, tag as NA for now to prepare for removal
ADNI2_geneExpression_Dx_values[407] <- NA
```




## Get diagnosis information for ADNIGO subjects

```{r}
adni_gene_expression_subjectInfo.split_phase$ADNIGO %>% head(10)
# since adnigo subjects already have viscode2 format for visit, can use visit column directly for matching with adnimerge dataset
# can match subjectID in gene expression data with PTID in adnimerge data
adnimerge_phaseSplit$ADNIGO %>% head(10)
```


```{r}
adni_gene_expression_subjectInfo.split_phase.ADNIGO <- adni_gene_expression_subjectInfo.split_phase$ADNIGO 
adni_gene_expression_subjectInfo.split_phase.ADNIGO %>% head

adnimerge_ADNIGO <- adnimerge_phaseSplit$ADNIGO
```


```{r}
# initialize empty vector to store diagnosis values
ADNIGO_geneExpression_Dx_values <- rep(0, nrow(adni_gene_expression_subjectInfo.split_phase.ADNIGO))
names(ADNIGO_geneExpression_Dx_values) <- rownames(adni_gene_expression_subjectInfo.split_phase.ADNIGO)

for (i in 1:length(ADNIGO_geneExpression_Dx_values)){
  
  subjectID_i <- adni_gene_expression_subjectInfo.split_phase.ADNIGO$SubjectID[i]
  visit_i <- adni_gene_expression_subjectInfo.split_phase.ADNIGO$Visit[i]
  
  # find index of subjectID in ADNIMERGE data
  adnimerge_subjectID_match_i <- which(adnimerge_ADNIGO$PTID == subjectID_i)
  # find index of visit in ADNIMERGE data
  adnimerge_visit_match_i <- which(adnimerge_ADNIGO$VISCODE == visit_i)
  # find intersection of index of subjectID and index of visit to give row number of diagnosis
  adnimerge_subjectID_visit_match_i <- intersect(adnimerge_subjectID_match_i, adnimerge_visit_match_i)
  # select row and diagnosis according to ADNIMERGE data
  subject_ADNIMERGE_Dx_i <- (adnimerge_ADNIGO[adnimerge_subjectID_visit_match_i,])$DX
  
  if(length(subject_ADNIMERGE_Dx_i) == 0){
    subject_ADNIMERGE_Dx_i = NA
  }
  
  # store diagnosis value into vector
  ADNIGO_geneExpression_Dx_values[i] = subject_ADNIMERGE_Dx_i
  
}
```

```{r}
ADNIGO_geneExpression_Dx_values %>% unique
# [1] "CN"       "MCI"      "Dementia" ""        

which(ADNIGO_geneExpression_Dx_values == "")
# subject_60 
#         22 

ADNIGO_geneExpression_Dx_values[22]

# check for diagnosis label for missing values
adni_gene_expression_subjectInfo.split_phase.ADNIGO[22,]
# subjectID 073_S_2182, visit m03
subject_adnimerge_match <- which(adnimerge_ADNIGO$PTID == "073_S_2182")
# [1] 752 753 754
adnimerge_ADNIGO[subject_adnimerge_match, c("PTID", "VISCODE", "DX_bl", "DX")]
# diagnosis at bl and m06 is MCI, so assign diagnosis MCI
ADNIGO_geneExpression_Dx_values[22] <- "MCI"
```


```{r}
# create vector of new visit codes (for ADNI2 and ADNIGO collated)
ADNIGO_geneExpression_VISCODE2_values <- adni_gene_expression_subjectInfo.split_phase.ADNIGO$Visit
names(ADNIGO_geneExpression_VISCODE2_values) <- rownames(adni_gene_expression_subjectInfo.split_phase.ADNIGO)
viscode_adni2_adnigo_geneExpression <- c(ADNI2_geneExpression_VISCODE2_values, ADNIGO_geneExpression_VISCODE2_values)
viscode_adni2_adnigo_geneExpression <- viscode_adni2_adnigo_geneExpression[
  match(rownames(adni_gene_expression_values), 
        names(viscode_adni2_adnigo_geneExpression))]
viscode_adni2_adnigo_geneExpression %>% table
#  bl m03 m06 m12 m36 m48 m60 m72 m84 
# 468   1   1   8   6 134 102  13   2 
viscode_adni2_adnigo_geneExpression %>% unique

# create vector of new diagnosis codes (for ADNI2 and ADNIGO collated
Dx_adni2_adnigo_geneExpression <- c(ADNI2_geneExpression_Dx_values, ADNIGO_geneExpression_Dx_values)
Dx_adni2_adnigo_geneExpression <- Dx_adni2_adnigo_geneExpression[
  match(rownames(adni_gene_expression_values), 
        names(Dx_adni2_adnigo_geneExpression))
]
Dx_adni2_adnigo_geneExpression %>% table
     #  CN Dementia      MCI 
     # 246      116      381 
Dx_adni2_adnigo_geneExpression %>% unique
```


```{r}
# add viscode2 and diagnosis values back to the subjectInfo dataframe
adni_gene_expression_subjectInfo_adnimerge_Dx <- adni_gene_expression_subjectInfo %>%
  mutate(VISCODE2 = viscode_adni2_adnigo_geneExpression,  
         ADNIMERGE_Dx = Dx_adni2_adnigo_geneExpression, .after = SubjectID)

adni_gene_expression_subjectInfo_adnimerge_Dx %>% head
```


## Add information about initial diagnosis, APOE genotype, age and sex

```{r}
# adding information about age (baased on age data provided by Prof Nho Kwangsik)
ADNI_geneExpression_age <- read.csv("../data/ADNI_raw_files/ADNI_mRNA_Age.csv")
ADNI_geneExpression_age %>% head

ADNI_geneExpression_age_ordered <- ADNI_geneExpression_age[match(adni_gene_expression_subjectInfo_adnimerge_Dx$SubjectID, ADNI_geneExpression_age$IID),]

adni_gene_expression_subjectInfo_adnimerge_Dx <- adni_gene_expression_subjectInfo_adnimerge_Dx %>%
  mutate(Age = ADNI_geneExpression_age_ordered$Age, .after=ADNIMERGE_Dx)

adni_gene_expression_subjectInfo_adnimerge_Dx %>% head
```


```{r}
# number of APOE4 alleles, age and sex from ADNIMERGE data sheet
adnimerge_selected_cols <- adnimerge[,c("COLPROT", "VISCODE", "PTID", "DX_bl", "PTGENDER", "APOE4", "PTRACCAT", "MMSE")]
adnimerge_selected_cols %>% head
```


```{r}
# initialize empty vectors
MMSE_vector <- rep(0, nrow(adni_gene_expression_subjectInfo_adnimerge_Dx))
DX_bl_vector <- rep(0, nrow(adni_gene_expression_subjectInfo_adnimerge_Dx))
PTGENDER_vector <- rep(0, nrow(adni_gene_expression_subjectInfo_adnimerge_Dx))
APOE4_vector <- rep(0, nrow(adni_gene_expression_subjectInfo_adnimerge_Dx))
PTRACCAT_vector <- rep(0, nrow(adni_gene_expression_subjectInfo_adnimerge_Dx))

for (i in 1:nrow(adni_gene_expression_subjectInfo_adnimerge_Dx)){
  
  # get information to match with adnimerge data
  phase_i <- adni_gene_expression_subjectInfo_adnimerge_Dx[i,]$Phase
  subjectID_i <- adni_gene_expression_subjectInfo_adnimerge_Dx[i,]$SubjectID
  viscode_i <- adni_gene_expression_subjectInfo_adnimerge_Dx[i,]$VISCODE2
  
  # match with adnimerge data
  phase_match_i <- which(adnimerge_selected_cols$COLPROT == phase_i)
  subjectID_match_i <- which(adnimerge_selected_cols$PTID == subjectID_i)
  viscode_match_i <- which(adnimerge_selected_cols$VISCODE == viscode_i)
  
  # for mmse, need to match phase, subjectID, and viscode
  row_intersect_phase_subjectID_viscode_i <- phase_match_i %>% 
    intersect(subjectID_match_i) %>% intersect(viscode_match_i)
  MMSE_i <- adnimerge_selected_cols[row_intersect_phase_subjectID_viscode_i,]$MMSE
  if (length(MMSE_i) == 0){
    MMSE_i = NA
  }
  MMSE_vector[i] <- MMSE_i
  
  # for the rest of the columns, only need a match for subjectID, use unique() to get only 1 value for each subject
  DX_bl_vector[i] <- adnimerge_selected_cols[subjectID_match_i,]$DX_bl %>% unique
  PTGENDER_vector[i] <- adnimerge_selected_cols[subjectID_match_i,]$PTGENDER %>% unique
  APOE4_vector[i] <- adnimerge_selected_cols[subjectID_match_i,]$APOE4 %>% unique
  PTRACCAT_vector[i] <- adnimerge_selected_cols[subjectID_match_i,]$PTRACCAT %>% unique
}
```


```{r}
which(is.na(MMSE_vector))
# [1]  60  73  79 124 210 361 386 487 540 636 679 683

adni_gene_expression_subjectInfo_adnimerge_Dx[60,]
# subject ID 073_S_2182, visit m03
# since MMSE at bl and m06 is 30, assume that MMSE at m03 is also 30
MMSE_vector[60] <- 30

adni_gene_expression_subjectInfo_adnimerge_Dx[73,]
# viscode2 is between bl and m06
# subject ID is 072_S_4102
# MMSE changes between bl and m06 from 30 to 25, so cannot assume MMSE at visit
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[79,]
# viscode2 is between bl and m06
# subject ID is 006_S_4192
# MMSE changes between bl and m06 from 19 to 22, so cannot assume MMSE at visit
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[124,]
# viscode2 is between bl and m06
# subject ID is 013_S_4268
# MMSE changes between bl and m05 from 29 to 30

adni_gene_expression_subjectInfo_adnimerge_Dx[210,]
# subject ID is 072_S_4103
# MMSE changes between bl and m05 from 29 to 28

adni_gene_expression_subjectInfo_adnimerge_Dx[361,]
# viscode2 is between sc and bl
# subject ID is 009_S_4564
# only MMSE available is at bl, MMSE = 27
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[386,]
# viscode2 is between bl and m06
# subject ID is 022_S_4266
# MMSE at bl and m06 = 30
MMSE_vector[386] <- 30

adni_gene_expression_subjectInfo_adnimerge_Dx[487,]
# Subject ID is 128_S_0135
# visit is m48
# no MMSE score available for visits before and after
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[540,]
# subject ID is 114_S_4379
# MMSE score at bl is 25, at m06 is 28
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[636,]
# subject ID is 094_S_4282
# MMSE score at bl is 23, at m06 is 21
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[679,]
# subject ID is 037_S_4432
# MMSE score at bl is 28, at m08 is 23
# leave MMSE as NA

adni_gene_expression_subjectInfo_adnimerge_Dx[683,]
# subject ID is 003_S_0907
# MMSE score consistently around 29 to 30
# assume MMSE is 30
MMSE_vector[683] <- 30
```



```{r}
# adni_gene_expression_subjectInfo_adnimerge_Dx %>% head

adni_gene_expression_subjectInfo_allInfo <- adni_gene_expression_subjectInfo_adnimerge_Dx %>% mutate(
  DX_bl = DX_bl_vector, 
  PTGENDER = PTGENDER_vector, 
  APOE4 = APOE4_vector, 
  Race = PTRACCAT_vector, 
  MMSE = MMSE_vector, 
  .after = Age
)

adni_gene_expression_subjectInfo_allInfo %>% head
```



```{r}
write.csv(adni_gene_expression_subjectInfo_allInfo, file = "20230215_adni_gene_expression_subjectInfo_Dx.csv")
```

```{r}
sessionInfo()
```


